app/com/linkedin/drelephant/mapreduce/TaskLevelAggregatedMetrics.java

```diff
@@ -0,0 +1,136 @@
+import org.apache.log4j.Logger;
+
+
+public class TaskLevelAggregatedMetrics {
+
+  private static final Logger logger = Logger.getLogger(TaskLevelAggregatedMetrics.class);
+
+  private long _delay = 0;
+  private long _wastedResources = 0;
+  private long _resourceUsage = 0;
+
+  private List<Long> finishTimes = new ArrayList<Long>();
+  private List<Long> durations = new ArrayList<Long>();
+
+  /**
+   * Returns the nth percentile finish job
+   * @param percentile The percentile of finish job to return
+   * @return The nth percentile finish job
+   */
+  public long getNthPercentileFinishTime(int percentile)
+  {
+    return Statistics.percentile(finishTimes, percentile);
+  }
+
+  /**
+   * Constructor for TaskLevelAggregatedMetrics
+   * @param taskData Array containing the task data for mappers and/or reducers
+   * @param containerSize The container size of the tasks
+   * @param idealStartTime The ideal start time for the task. For mappers it is the submit time, for
+   *                       reducers, it is the time when the number of completed maps become more than
+   *                       the slow start time.
+   */
+  public TaskLevelAggregatedMetrics(MapReduceTaskData[] taskData, long containerSize, long idealStartTime) {
+    compute(taskData, containerSize, idealStartTime);
+  }
+
+  /**
+   * Returns the overall delay for the tasks.
+   * @return The delay of the tasks.
+   */
+  public long getDelay() {
+    return _delay;
+  }
+
+  /**
+   * Retruns the resources wasted by all the tasks in MB Seconds
+   * @return The wasted resources of all the tasks in MB Seconds
+   */
+  public long getWastedResources() {
+    return _wastedResources;
+  }
+
+  /**
+   * Returns the resource used by all the tasks in MB Seconds
+   * @return The total resources used by all tasks in MB Seconds
+   */
+  public long getResourceUsage() {
+    return _resourceUsage;
+  }
+
+  /**
+   * Computes the aggregated metrics -> peakMemory, delay, total task duration, wasted resources and memory usage.
+   * @param taskDatas
+   * @param containerSize
+   * @param idealStartTime
+   */
+  private void compute(MapReduceTaskData[] taskDatas, long containerSize, long idealStartTime) {
+
+    long peakMemoryNeed = 0;
+    long taskFinishTimeMax = 0;
+    long taskDurationMax = 0;
+
+    for (MapReduceTaskData taskData: taskDatas) {
+      long taskMemory = taskData.getCounters().get(MapReduceCounterData.CounterName.PHYSICAL_MEMORY_BYTES)/Statistics.MB; // MB
+      long taskVM = taskData.getCounters().get(MapReduceCounterData.CounterName.VIRTUAL_MEMORY_BYTES)/ Statistics.MB; // MB
+      long taskDuration = taskData.getFinishTimeMs() - taskData.getStartTimeMs(); // Milliseconds
+      long taskCost =  (containerSize) * (taskDuration / Statistics.SECOND_IN_MS); // MB Seconds
+
+      durations.add(taskDuration);
+      finishTimes.add(taskData.getFinishTimeMs());
+
+      //peak Memory usage
+      long memoryRequiredForVM = (long) (taskVM/2.1);
```

Comment:
Use a constant and also add a comment saying this is because that is the cluster wide config we have. Actually, we should make it a configurable parameter as other clusters might have a different setting for this.